lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -230,6 +246,27 @@ public function execute($params = null)
         return true;
     }
 
+    /**
+     * Send BLOB type streams using mysqli::send_long_data
+     */
+    private function _sendLongData()
+    {
+        foreach ($this->_longData as $paramNr => $resource) {
+            if (is_resource($resource) && get_resource_type($resource) === 'stream') {
+                while (! feof($resource)) {
+                    $longData = fread($resource, 8192);
```

Comment:
`fread()` can return `false` in case of a failure. It should be handled before trying to send the data.

Reply:
:+1: